crypto/ed25519.go

```diff
@@ -74,10 +75,7 @@ func GeneratePrivateKey() (PrivateKey, error) {
 // PublicKey returns a PublicKey associated with the Ed25519 PrivateKey p.
 // The PublicKey is the last 32 bytes of p.
 func (p PrivateKey) PublicKey() PublicKey {
-	rpk := p[32:] // privateKey == private|public
-	var pk PublicKey
-	copy(pk[:], rpk)
-	return pk
+	return PublicKey(p[32:])
```

Comment:
Very nice...not sure why I didn't do this in the first place.

You should use `PublicKeyLen` here instead of 32. I also think you should keep the commend that explains why we go 32 in.

Reply:
Done! Thank you!